### ISSUE TITLE
File uploads on https://staging-azure.civiform.dev/ don't work

### ISSUE THREAD

**Issue by clouser2** [civiform/civiform#2718]
**Describe the bug**
File uploads on our Azure staging instance throw a CORS error.

**To Reproduce**
1. Go to https://staging-azure.civiform.dev/
2. Apply to a program with a file upload
3. Choose a file and click "Upload"
4. Nothing happens

```
Access to XMLHttpRequest at 'https://stagingqxzox.blob.core.windows.net/files/...' from origin 'https://staging-azure.civiform.dev' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**Expected behavior**
File uploads should work

**Root cause**
Server-side, we generate the parameters used for the client-side JavaScript to use when making a blob upload request to Azure. One of these is setting relevant CORS headers, where we do set an `Access-Control-Allow-Origin` header ([code](https://github.com/seattle-uat/civiform/blob/509ec3b16ac2d5ae4d5373589423db8719014574/server/app/services/cloud/azure/BlobStorage.java#L133)). We set this to the domain configured as the "base_url" configuration value ([code](https://github.com/seattle-uat/civiform/blob/509ec3b16ac2d5ae4d5373589423db8719014574/server/app/services/cloud/azure/BlobStorage.java#L173)).

This consults the `BASE_URL` environment variable and defaults to `http://localhost:9000` ([code](https://github.com/seattle-uat/civiform/blob/509ec3b16ac2d5ae4d5373589423db8719014574/server/conf/application.conf#L107)). As far as I can tell, we don't set this environment variable anywhere in our Staging terraform config. The closest thing I can find is the `custom_hostname` property ([code](https://github.com/seattle-uat/civiform/blob/509ec3b16ac2d5ae4d5373589423db8719014574/cloud/azure/templates/azure_saml_ses/variables.tf#L69-L73)). The docs for this property claim that it's used for CORS, but I don't ever see the value being wired up to a `BASE_URL` environment variable.

I believe this is the reason file uploads aren't working for our Azure staging environment. Based on this, I can't see how they ever would have worked on this instance. Once we have staging probers, it'll be a great way to catch these types of issues sooner.

**Comment by goldblatt** [comment]
cc @azizova-leyla

**Comment by goldblatt** [comment]
ahh yeah so somewhat interestingly the deploy system sets the BASE_URL via the azure::slot_setting command!

**Comment by goldblatt** [comment]
and just checked it is set to https://staging-azure.civiform.dev

**Comment by clouser2** [comment]
This now works. @goldblatt had changed some unrelated settings within the Azure portal. We don't have a good idea what's happening here or what fixed it. We're not going to invest future effort here.

**Comment by glorialiou** [comment]
Prioritized as P2 as no entities are using Azure (staging) at the moment.

**Comment by gwendolyngoetz** [comment]
@lagosr-google fyi, probably related to what you're working on.

**Comment by dkatzz** [comment]
Closing this as a duplicate of https://github.com/civiform/civiform/issues/8108
